# Machine Learning Engineer Nanodegree

## Capstone Project - Smile Detector


---

In this project, CNNs are used to build models to detect if the person in the image is smiling or not.
CelibA dataset is used for this purpose - http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html
The data for this project is downloaded from Kaggle - https://www.kaggle.com/jessicali9530/celeba-dataset
Here, transfer learning with a pretrained model VGG19 is used with random weights and SGD optimizer.

In [15]:
IMG_H=218
IMG_W=178
IMG_D=3

In the code cell below, we populate a few variables through the use of the load_files function from the scikit-learn library:

train_files, valid_files, test_files - numpy arrays containing file paths to images train_targets, valid_targets, test_targets - numpy arrays containing onehot-encoded classification labels smile_names - list of string-valued smile categories for translating labels.

In [2]:
from sklearn.datasets import load_files       
from keras.utils import np_utils
import numpy as np
from glob import glob

# define function to load train, test, and validation datasets
def load_dataset(path):
    data = load_files(path, load_content=False)
    smile_files = np.array(data['filenames'])
    smile_targets = np_utils.to_categorical(np.array(data['target']), 2)
    return smile_files, smile_targets

# load train, test, and validation datasets
train_files, train_targets = load_dataset('input/dataset/train')
valid_files, valid_targets = load_dataset('input/dataset/validate')
test_files, test_targets = load_dataset('input/dataset/test')

smile_names = [item[:-1] for item in sorted(glob("input/dataset/train/*/"))]

# print statistics about the dataset
print('There are %d total smile categories.' % len(smile_names))
print('There are %s total smile images.\n' % len(np.hstack([train_files, valid_files, test_files])))
print('There are %d training smile images.' % len(train_files))
print('There are %d validation smile images.' % len(valid_files))
print('There are %d test smile images.'% len(test_files))

Using TensorFlow backend.


There are 2 total smile categories.
There are 15000 total smile images.

There are 10000 training smile images.
There are 2500 validation smile images.
There are 2500 test smile images.


When using TensorFlow as backend, Keras CNNs require a 4D array as input. Below is the function for the same. Also, preprocess input is used from VGG19 which is the chosen pre-trained model

In [3]:
from tqdm import tqdm
from keras.applications.vgg19 import VGG19
from keras.preprocessing import image
from keras.applications.vgg19 import preprocess_input

def path_to_tensor(img_path):
    # loads RGB image as PIL.Image.Image type
    img = image.load_img(img_path, target_size=(IMG_H, IMG_W))
    # convert PIL.Image.Image type to 3D tensor with shape (IMG_H, IMG_W, IMG_D)
    x = image.img_to_array(img)
    # convert 3D tensor to 4D tensor with shape (1, IMG_H, IMG_W, IMG_D) and return 4D tensor
    x = preprocess_input(x)
    return np.expand_dims(x, axis=0)
    
def paths_to_tensor(img_paths):
    list_of_tensors = [path_to_tensor(img_path) for img_path in tqdm(img_paths)]
    return np.vstack(list_of_tensors)

In [4]:
from PIL import ImageFile                            
ImageFile.LOAD_TRUNCATED_IMAGES = True                 

# pre-process the data for Keras
train_tensors = paths_to_tensor(train_files)
valid_tensors = paths_to_tensor(valid_files)
test_tensors = paths_to_tensor(test_files)

100%|██████████| 2500/2500 [01:17<00:00, 32.08it/s]


### Understanding the VGG19 model.

In [5]:
orig_model = VGG19(weights='imagenet',
                  include_top=False,
                    input_shape=(IMG_H, IMG_W, IMG_D))
print("number of layers:", len(orig_model.layers))
orig_model.summary()

number of layers: 22
Model: "vgg19"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
input_1 (InputLayer)         (None, 218, 178, 3)       0         
_________________________________________________________________
block1_conv1 (Conv2D)        (None, 218, 178, 64)      1792      
_________________________________________________________________
block1_conv2 (Conv2D)        (None, 218, 178, 64)      36928     
_________________________________________________________________
block1_pool (MaxPooling2D)   (None, 109, 89, 64)       0         
_________________________________________________________________
block2_conv1 (Conv2D)        (None, 109, 89, 128)      73856     
_________________________________________________________________
block2_conv2 (Conv2D)        (None, 109, 89, 128)      147584    
_________________________________________________________________
block2_pool (MaxPooling2D)   (None, 54, 


After compiling the below mentioned model with rmsprop optimizer, trying to explore with SGD (Stochastic Gradient Descent). Started with a very low learning rate of 0.0001 and momentum 0.9

In [6]:
from keras.layers import GlobalAveragePooling2D, Dense, Dropout
from keras.models import Model
from keras.optimizers import SGD

newLayer = orig_model.output
newLayer = GlobalAveragePooling2D()(newLayer)
newLayer = Dense(1024, activation='relu')(newLayer)
newLayer = Dropout(0.25)(newLayer)
newLayer = Dense(512, activation="relu")(newLayer)
newLayer = Dropout(0.25)(newLayer);
predictions = Dense(2,activation='softmax')(newLayer)
newModel = Model(inputs=orig_model.input,outputs=predictions)

# train only the top layers (which were randomly initialized)
# i.e. freeze all convolutional InceptionV3 layers
for layer in newModel.layers[:22]:
    layer.trainable = False


In [7]:
# compile the model (should be done *after* setting layers to non-trainable)
newModel.compile(optimizer=SGD(lr=0.0001, momentum=0.9), loss='categorical_crossentropy', metrics=['accuracy'])

In [8]:
newModel.summary()

Model: "model_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
input_1 (InputLayer)         (None, 218, 178, 3)       0         
_________________________________________________________________
block1_conv1 (Conv2D)        (None, 218, 178, 64)      1792      
_________________________________________________________________
block1_conv2 (Conv2D)        (None, 218, 178, 64)      36928     
_________________________________________________________________
block1_pool (MaxPooling2D)   (None, 109, 89, 64)       0         
_________________________________________________________________
block2_conv1 (Conv2D)        (None, 109, 89, 128)      73856     
_________________________________________________________________
block2_conv2 (Conv2D)        (None, 109, 89, 128)      147584    
_________________________________________________________________
block2_pool (MaxPooling2D)   (None, 54, 44, 128)       0   

In [9]:
trainable = 0
for layer in newModel.layers:
    if (layer.trainable == True):
            trainable = trainable + 1;
print ("trainable Layers = ", trainable)            

trainable Layers =  6


In [10]:
from keras.callbacks import ModelCheckpoint
checkpointer = ModelCheckpoint(filepath='input/saved_models/transfer_models_SGD.weights.best.hdf5',
                               verbose=1,save_best_only=True)

#### Fit the model

In [11]:
NUM_EPOCHS=20
BATCH_SIZE=32
newModel.fit(train_tensors, train_targets, 
          validation_data=(valid_tensors, valid_targets),
          epochs=NUM_EPOCHS, batch_size=BATCH_SIZE, callbacks=[checkpointer], verbose=1)

Train on 10000 samples, validate on 2500 samples
Epoch 1/20


 5600/10000 [===============>..............] - ETA: 1:41:48 - loss: 1.2278 - accuracy: 0.500 - ETA: 1:24:33 - loss: 1.4656 - accuracy: 0.437 - ETA: 1:19:30 - loss: 1.4503 - accuracy: 0.447 - ETA: 1:16:40 - loss: 1.5574 - accuracy: 0.445 - ETA: 1:14:31 - loss: 1.4226 - accuracy: 0.487 - ETA: 1:13:06 - loss: 1.5547 - accuracy: 0.453 - ETA: 1:12:04 - loss: 1.5212 - accuracy: 0.446 - ETA: 1:11:19 - loss: 1.5544 - accuracy: 0.460 - ETA: 1:10:31 - loss: 1.4991 - accuracy: 0.472 - ETA: 1:09:54 - loss: 1.5241 - accuracy: 0.456 - ETA: 1:09:17 - loss: 1.4949 - accuracy: 0.465 - ETA: 1:08:42 - loss: 1.4903 - accuracy: 0.460 - ETA: 1:08:01 - loss: 1.4768 - accuracy: 0.461 - ETA: 1:07:28 - loss: 1.4439 - accuracy: 0.473 - ETA: 1:07:05 - loss: 1.4548 - accuracy: 0.470 - ETA: 1:06:39 - loss: 1.4456 - accuracy: 0.476 - ETA: 1:06:13 - loss: 1.4215 - accuracy: 0.477 - ETA: 1:05:43 - loss: 1.4226 - accuracy: 0.475 - ETA: 1:05:20 - loss: 1.4047 - accuracy: 0.480 - ETA: 1:05:02 - loss: 1.4002 - accuracy: 0

10000/10000 [==============================] - ETA: 29:44 - loss: 0.9615 - accuracy: 0.606 - ETA: 29:30 - loss: 0.9619 - accuracy: 0.606 - ETA: 29:17 - loss: 0.9605 - accuracy: 0.606 - ETA: 29:04 - loss: 0.9600 - accuracy: 0.606 - ETA: 28:51 - loss: 0.9589 - accuracy: 0.607 - ETA: 28:37 - loss: 0.9587 - accuracy: 0.607 - ETA: 28:25 - loss: 0.9585 - accuracy: 0.607 - ETA: 28:12 - loss: 0.9586 - accuracy: 0.607 - ETA: 27:59 - loss: 0.9584 - accuracy: 0.607 - ETA: 27:45 - loss: 0.9564 - accuracy: 0.608 - ETA: 27:32 - loss: 0.9553 - accuracy: 0.608 - ETA: 27:19 - loss: 0.9525 - accuracy: 0.609 - ETA: 27:06 - loss: 0.9523 - accuracy: 0.610 - ETA: 26:53 - loss: 0.9525 - accuracy: 0.610 - ETA: 26:40 - loss: 0.9513 - accuracy: 0.610 - ETA: 26:27 - loss: 0.9519 - accuracy: 0.609 - ETA: 26:14 - loss: 0.9498 - accuracy: 0.610 - ETA: 26:02 - loss: 0.9495 - accuracy: 0.610 - ETA: 25:48 - loss: 0.9490 - accuracy: 0.610 - ETA: 25:35 - loss: 0.9475 - accuracy: 0.610 - ETA: 25:22 - loss: 0.9465 - accur

 5632/10000 [===============>..............] - ETA: 1:08:58 - loss: 0.5119 - accuracy: 0.750 - ETA: 1:07:54 - loss: 0.6015 - accuracy: 0.703 - ETA: 1:06:45 - loss: 0.5699 - accuracy: 0.708 - ETA: 1:06:24 - loss: 0.6122 - accuracy: 0.703 - ETA: 1:05:53 - loss: 0.6325 - accuracy: 0.687 - ETA: 1:05:46 - loss: 0.6581 - accuracy: 0.682 - ETA: 1:05:32 - loss: 0.6509 - accuracy: 0.683 - ETA: 1:05:00 - loss: 0.6373 - accuracy: 0.687 - ETA: 1:04:45 - loss: 0.6289 - accuracy: 0.684 - ETA: 1:04:32 - loss: 0.6660 - accuracy: 0.671 - ETA: 1:04:39 - loss: 0.6645 - accuracy: 0.670 - ETA: 1:04:24 - loss: 0.6475 - accuracy: 0.677 - ETA: 1:04:07 - loss: 0.6711 - accuracy: 0.668 - ETA: 1:03:54 - loss: 0.6869 - accuracy: 0.665 - ETA: 1:03:40 - loss: 0.7214 - accuracy: 0.654 - ETA: 1:03:22 - loss: 0.7141 - accuracy: 0.662 - ETA: 1:03:07 - loss: 0.7097 - accuracy: 0.663 - ETA: 1:02:53 - loss: 0.7167 - accuracy: 0.659 - ETA: 1:02:38 - loss: 0.7127 - accuracy: 0.666 - ETA: 1:02:25 - loss: 0.7063 - accuracy: 0

10000/10000 [==============================] - ETA: 29:05 - loss: 0.6850 - accuracy: 0.672 - ETA: 28:53 - loss: 0.6849 - accuracy: 0.672 - ETA: 28:40 - loss: 0.6852 - accuracy: 0.672 - ETA: 28:27 - loss: 0.6841 - accuracy: 0.672 - ETA: 28:14 - loss: 0.6828 - accuracy: 0.673 - ETA: 28:01 - loss: 0.6819 - accuracy: 0.673 - ETA: 27:49 - loss: 0.6810 - accuracy: 0.673 - ETA: 27:36 - loss: 0.6810 - accuracy: 0.673 - ETA: 27:23 - loss: 0.6810 - accuracy: 0.673 - ETA: 27:11 - loss: 0.6799 - accuracy: 0.674 - ETA: 26:58 - loss: 0.6805 - accuracy: 0.674 - ETA: 26:45 - loss: 0.6804 - accuracy: 0.674 - ETA: 26:33 - loss: 0.6791 - accuracy: 0.674 - ETA: 26:20 - loss: 0.6782 - accuracy: 0.675 - ETA: 26:07 - loss: 0.6790 - accuracy: 0.675 - ETA: 25:54 - loss: 0.6799 - accuracy: 0.674 - ETA: 25:42 - loss: 0.6808 - accuracy: 0.673 - ETA: 25:29 - loss: 0.6818 - accuracy: 0.672 - ETA: 25:16 - loss: 0.6805 - accuracy: 0.673 - ETA: 25:03 - loss: 0.6812 - accuracy: 0.673 - ETA: 24:51 - loss: 0.6815 - accur

 5632/10000 [===============>..............] - ETA: 1:06:58 - loss: 0.5110 - accuracy: 0.687 - ETA: 1:06:03 - loss: 0.6257 - accuracy: 0.625 - ETA: 1:05:10 - loss: 0.7212 - accuracy: 0.583 - ETA: 1:04:48 - loss: 0.6901 - accuracy: 0.617 - ETA: 1:04:47 - loss: 0.7284 - accuracy: 0.606 - ETA: 1:04:38 - loss: 0.7368 - accuracy: 0.609 - ETA: 1:04:23 - loss: 0.7479 - accuracy: 0.611 - ETA: 1:04:10 - loss: 0.7112 - accuracy: 0.636 - ETA: 1:04:03 - loss: 0.7054 - accuracy: 0.649 - ETA: 1:03:58 - loss: 0.6963 - accuracy: 0.653 - ETA: 1:03:43 - loss: 0.6823 - accuracy: 0.659 - ETA: 1:03:32 - loss: 0.6734 - accuracy: 0.664 - ETA: 1:03:20 - loss: 0.6555 - accuracy: 0.675 - ETA: 1:03:04 - loss: 0.6405 - accuracy: 0.685 - ETA: 1:02:52 - loss: 0.6334 - accuracy: 0.691 - ETA: 1:02:38 - loss: 0.6354 - accuracy: 0.691 - ETA: 1:02:19 - loss: 0.6267 - accuracy: 0.698 - ETA: 1:02:06 - loss: 0.6197 - accuracy: 0.703 - ETA: 1:01:55 - loss: 0.6161 - accuracy: 0.705 - ETA: 1:01:38 - loss: 0.6131 - accuracy: 0

10000/10000 [==============================] - ETA: 28:46 - loss: 0.6170 - accuracy: 0.693 - ETA: 28:33 - loss: 0.6187 - accuracy: 0.692 - ETA: 28:21 - loss: 0.6193 - accuracy: 0.692 - ETA: 28:08 - loss: 0.6216 - accuracy: 0.691 - ETA: 27:55 - loss: 0.6222 - accuracy: 0.691 - ETA: 27:43 - loss: 0.6225 - accuracy: 0.691 - ETA: 27:30 - loss: 0.6231 - accuracy: 0.691 - ETA: 27:17 - loss: 0.6236 - accuracy: 0.690 - ETA: 27:04 - loss: 0.6234 - accuracy: 0.691 - ETA: 26:52 - loss: 0.6240 - accuracy: 0.691 - ETA: 26:39 - loss: 0.6232 - accuracy: 0.692 - ETA: 26:26 - loss: 0.6226 - accuracy: 0.692 - ETA: 26:13 - loss: 0.6224 - accuracy: 0.692 - ETA: 26:01 - loss: 0.6231 - accuracy: 0.691 - ETA: 25:48 - loss: 0.6225 - accuracy: 0.692 - ETA: 25:35 - loss: 0.6233 - accuracy: 0.691 - ETA: 25:23 - loss: 0.6236 - accuracy: 0.691 - ETA: 25:10 - loss: 0.6231 - accuracy: 0.691 - ETA: 24:57 - loss: 0.6227 - accuracy: 0.691 - ETA: 24:45 - loss: 0.6220 - accuracy: 0.691 - ETA: 24:32 - loss: 0.6220 - accur

 5632/10000 [===============>..............] - ETA: 1:05:21 - loss: 0.5944 - accuracy: 0.750 - ETA: 1:04:15 - loss: 0.5408 - accuracy: 0.765 - ETA: 1:04:07 - loss: 0.6271 - accuracy: 0.718 - ETA: 1:04:03 - loss: 0.6425 - accuracy: 0.679 - ETA: 1:03:57 - loss: 0.6296 - accuracy: 0.668 - ETA: 1:03:37 - loss: 0.5971 - accuracy: 0.682 - ETA: 1:03:33 - loss: 0.6047 - accuracy: 0.660 - ETA: 1:03:30 - loss: 0.5860 - accuracy: 0.675 - ETA: 1:03:19 - loss: 0.5912 - accuracy: 0.673 - ETA: 1:03:21 - loss: 0.6173 - accuracy: 0.665 - ETA: 1:03:07 - loss: 0.6219 - accuracy: 0.659 - ETA: 1:03:01 - loss: 0.6169 - accuracy: 0.666 - ETA: 1:02:51 - loss: 0.6020 - accuracy: 0.675 - ETA: 1:02:40 - loss: 0.6086 - accuracy: 0.667 - ETA: 1:02:28 - loss: 0.6005 - accuracy: 0.672 - ETA: 1:02:12 - loss: 0.6101 - accuracy: 0.673 - ETA: 1:02:01 - loss: 0.6132 - accuracy: 0.671 - ETA: 1:01:48 - loss: 0.6094 - accuracy: 0.673 - ETA: 1:01:37 - loss: 0.6079 - accuracy: 0.679 - ETA: 1:01:23 - loss: 0.6143 - accuracy: 0

10000/10000 [==============================] - ETA: 29:02 - loss: 0.5848 - accuracy: 0.701 - ETA: 28:49 - loss: 0.5845 - accuracy: 0.701 - ETA: 28:36 - loss: 0.5835 - accuracy: 0.702 - ETA: 28:23 - loss: 0.5843 - accuracy: 0.702 - ETA: 28:10 - loss: 0.5854 - accuracy: 0.702 - ETA: 27:58 - loss: 0.5851 - accuracy: 0.702 - ETA: 27:45 - loss: 0.5850 - accuracy: 0.702 - ETA: 27:32 - loss: 0.5855 - accuracy: 0.701 - ETA: 27:19 - loss: 0.5862 - accuracy: 0.701 - ETA: 27:06 - loss: 0.5861 - accuracy: 0.701 - ETA: 26:54 - loss: 0.5861 - accuracy: 0.701 - ETA: 26:41 - loss: 0.5871 - accuracy: 0.700 - ETA: 26:28 - loss: 0.5865 - accuracy: 0.700 - ETA: 26:16 - loss: 0.5872 - accuracy: 0.700 - ETA: 26:03 - loss: 0.5867 - accuracy: 0.700 - ETA: 25:50 - loss: 0.5871 - accuracy: 0.700 - ETA: 25:37 - loss: 0.5881 - accuracy: 0.700 - ETA: 25:25 - loss: 0.5877 - accuracy: 0.700 - ETA: 25:12 - loss: 0.5878 - accuracy: 0.700 - ETA: 25:00 - loss: 0.5866 - accuracy: 0.701 - ETA: 24:47 - loss: 0.5860 - accur

 5632/10000 [===============>..............] - ETA: 1:06:31 - loss: 0.5579 - accuracy: 0.750 - ETA: 1:06:31 - loss: 0.5149 - accuracy: 0.765 - ETA: 1:05:43 - loss: 0.4586 - accuracy: 0.812 - ETA: 1:05:46 - loss: 0.4600 - accuracy: 0.804 - ETA: 1:05:33 - loss: 0.4560 - accuracy: 0.800 - ETA: 1:05:35 - loss: 0.5164 - accuracy: 0.765 - ETA: 1:05:10 - loss: 0.5002 - accuracy: 0.772 - ETA: 1:04:46 - loss: 0.5049 - accuracy: 0.761 - ETA: 1:04:27 - loss: 0.5005 - accuracy: 0.763 - ETA: 1:04:09 - loss: 0.5106 - accuracy: 0.759 - ETA: 1:04:16 - loss: 0.5107 - accuracy: 0.761 - ETA: 1:04:15 - loss: 0.5135 - accuracy: 0.765 - ETA: 1:03:54 - loss: 0.5166 - accuracy: 0.759 - ETA: 1:03:37 - loss: 0.5303 - accuracy: 0.756 - ETA: 1:03:29 - loss: 0.5224 - accuracy: 0.762 - ETA: 1:03:22 - loss: 0.5233 - accuracy: 0.759 - ETA: 1:03:12 - loss: 0.5195 - accuracy: 0.762 - ETA: 1:02:54 - loss: 0.5325 - accuracy: 0.751 - ETA: 1:02:38 - loss: 0.5362 - accuracy: 0.750 - ETA: 1:02:24 - loss: 0.5434 - accuracy: 0

10000/10000 [==============================] - ETA: 28:53 - loss: 0.5843 - accuracy: 0.709 - ETA: 28:40 - loss: 0.5840 - accuracy: 0.709 - ETA: 28:27 - loss: 0.5842 - accuracy: 0.709 - ETA: 28:15 - loss: 0.5835 - accuracy: 0.709 - ETA: 28:02 - loss: 0.5835 - accuracy: 0.709 - ETA: 27:50 - loss: 0.5826 - accuracy: 0.709 - ETA: 27:37 - loss: 0.5832 - accuracy: 0.708 - ETA: 27:24 - loss: 0.5841 - accuracy: 0.708 - ETA: 27:11 - loss: 0.5827 - accuracy: 0.709 - ETA: 26:59 - loss: 0.5823 - accuracy: 0.709 - ETA: 26:46 - loss: 0.5819 - accuracy: 0.709 - ETA: 26:33 - loss: 0.5814 - accuracy: 0.710 - ETA: 26:20 - loss: 0.5808 - accuracy: 0.710 - ETA: 26:08 - loss: 0.5800 - accuracy: 0.711 - ETA: 25:55 - loss: 0.5801 - accuracy: 0.711 - ETA: 25:42 - loss: 0.5803 - accuracy: 0.710 - ETA: 25:30 - loss: 0.5793 - accuracy: 0.711 - ETA: 25:17 - loss: 0.5797 - accuracy: 0.711 - ETA: 25:04 - loss: 0.5790 - accuracy: 0.711 - ETA: 24:52 - loss: 0.5794 - accuracy: 0.711 - ETA: 24:39 - loss: 0.5792 - accur

 5632/10000 [===============>..............] - ETA: 1:05:27 - loss: 0.5167 - accuracy: 0.687 - ETA: 1:05:22 - loss: 0.5350 - accuracy: 0.687 - ETA: 1:05:20 - loss: 0.5628 - accuracy: 0.697 - ETA: 1:05:03 - loss: 0.5157 - accuracy: 0.718 - ETA: 1:04:42 - loss: 0.5224 - accuracy: 0.706 - ETA: 1:04:20 - loss: 0.6021 - accuracy: 0.671 - ETA: 1:04:14 - loss: 0.5849 - accuracy: 0.669 - ETA: 1:04:04 - loss: 0.5755 - accuracy: 0.675 - ETA: 1:03:57 - loss: 0.5717 - accuracy: 0.673 - ETA: 1:03:38 - loss: 0.5660 - accuracy: 0.678 - ETA: 1:03:25 - loss: 0.5742 - accuracy: 0.679 - ETA: 1:03:13 - loss: 0.5710 - accuracy: 0.687 - ETA: 1:03:03 - loss: 0.5800 - accuracy: 0.687 - ETA: 1:02:51 - loss: 0.5764 - accuracy: 0.694 - ETA: 1:02:37 - loss: 0.5597 - accuracy: 0.706 - ETA: 1:02:34 - loss: 0.5702 - accuracy: 0.705 - ETA: 1:02:22 - loss: 0.5723 - accuracy: 0.702 - ETA: 1:02:08 - loss: 0.5715 - accuracy: 0.703 - ETA: 1:01:55 - loss: 0.5687 - accuracy: 0.705 - ETA: 1:01:38 - loss: 0.5623 - accuracy: 0

10000/10000 [==============================] - ETA: 29:00 - loss: 0.5482 - accuracy: 0.724 - ETA: 28:47 - loss: 0.5483 - accuracy: 0.724 - ETA: 28:35 - loss: 0.5483 - accuracy: 0.724 - ETA: 28:22 - loss: 0.5482 - accuracy: 0.723 - ETA: 28:10 - loss: 0.5489 - accuracy: 0.723 - ETA: 27:58 - loss: 0.5489 - accuracy: 0.722 - ETA: 27:45 - loss: 0.5492 - accuracy: 0.722 - ETA: 27:33 - loss: 0.5492 - accuracy: 0.722 - ETA: 27:20 - loss: 0.5501 - accuracy: 0.721 - ETA: 27:08 - loss: 0.5517 - accuracy: 0.720 - ETA: 26:55 - loss: 0.5524 - accuracy: 0.720 - ETA: 26:43 - loss: 0.5531 - accuracy: 0.719 - ETA: 26:30 - loss: 0.5533 - accuracy: 0.719 - ETA: 26:18 - loss: 0.5531 - accuracy: 0.719 - ETA: 26:06 - loss: 0.5533 - accuracy: 0.719 - ETA: 25:53 - loss: 0.5534 - accuracy: 0.719 - ETA: 25:41 - loss: 0.5539 - accuracy: 0.719 - ETA: 25:28 - loss: 0.5544 - accuracy: 0.719 - ETA: 25:15 - loss: 0.5546 - accuracy: 0.719 - ETA: 25:02 - loss: 0.5549 - accuracy: 0.719 - ETA: 24:49 - loss: 0.5552 - accur

 5632/10000 [===============>..............] - ETA: 1:14:29 - loss: 0.4590 - accuracy: 0.843 - ETA: 1:11:22 - loss: 0.5357 - accuracy: 0.765 - ETA: 1:10:09 - loss: 0.4912 - accuracy: 0.760 - ETA: 1:09:36 - loss: 0.4832 - accuracy: 0.773 - ETA: 1:09:36 - loss: 0.4730 - accuracy: 0.775 - ETA: 1:09:07 - loss: 0.4756 - accuracy: 0.770 - ETA: 1:08:41 - loss: 0.4788 - accuracy: 0.763 - ETA: 1:08:07 - loss: 0.4809 - accuracy: 0.765 - ETA: 1:07:40 - loss: 0.5098 - accuracy: 0.750 - ETA: 1:07:13 - loss: 0.5092 - accuracy: 0.753 - ETA: 1:06:46 - loss: 0.5219 - accuracy: 0.741 - ETA: 1:06:22 - loss: 0.5318 - accuracy: 0.734 - ETA: 1:06:01 - loss: 0.5329 - accuracy: 0.733 - ETA: 1:05:42 - loss: 0.5280 - accuracy: 0.736 - ETA: 1:05:24 - loss: 0.5311 - accuracy: 0.743 - ETA: 1:05:07 - loss: 0.5299 - accuracy: 0.744 - ETA: 1:04:44 - loss: 0.5316 - accuracy: 0.740 - ETA: 1:04:27 - loss: 0.5351 - accuracy: 0.736 - ETA: 1:04:09 - loss: 0.5277 - accuracy: 0.743 - ETA: 1:03:49 - loss: 0.5368 - accuracy: 0

10000/10000 [==============================] - ETA: 29:04 - loss: 0.5460 - accuracy: 0.728 - ETA: 28:51 - loss: 0.5456 - accuracy: 0.728 - ETA: 28:38 - loss: 0.5452 - accuracy: 0.729 - ETA: 28:26 - loss: 0.5451 - accuracy: 0.729 - ETA: 28:13 - loss: 0.5457 - accuracy: 0.729 - ETA: 28:00 - loss: 0.5447 - accuracy: 0.729 - ETA: 27:47 - loss: 0.5444 - accuracy: 0.730 - ETA: 27:34 - loss: 0.5444 - accuracy: 0.730 - ETA: 27:21 - loss: 0.5439 - accuracy: 0.730 - ETA: 27:08 - loss: 0.5439 - accuracy: 0.730 - ETA: 26:55 - loss: 0.5432 - accuracy: 0.730 - ETA: 26:42 - loss: 0.5432 - accuracy: 0.731 - ETA: 26:29 - loss: 0.5430 - accuracy: 0.730 - ETA: 26:16 - loss: 0.5438 - accuracy: 0.730 - ETA: 26:03 - loss: 0.5432 - accuracy: 0.731 - ETA: 25:50 - loss: 0.5426 - accuracy: 0.731 - ETA: 25:37 - loss: 0.5428 - accuracy: 0.731 - ETA: 25:25 - loss: 0.5429 - accuracy: 0.730 - ETA: 25:12 - loss: 0.5424 - accuracy: 0.731 - ETA: 24:59 - loss: 0.5419 - accuracy: 0.731 - ETA: 24:46 - loss: 0.5418 - accur

 5632/10000 [===============>..............] - ETA: 1:10:24 - loss: 0.7713 - accuracy: 0.562 - ETA: 1:08:25 - loss: 0.7070 - accuracy: 0.562 - ETA: 1:07:23 - loss: 0.6263 - accuracy: 0.614 - ETA: 1:06:33 - loss: 0.5636 - accuracy: 0.671 - ETA: 1:06:08 - loss: 0.5774 - accuracy: 0.681 - ETA: 1:05:50 - loss: 0.5829 - accuracy: 0.692 - ETA: 1:05:34 - loss: 0.5777 - accuracy: 0.692 - ETA: 1:05:25 - loss: 0.5629 - accuracy: 0.699 - ETA: 1:05:12 - loss: 0.5501 - accuracy: 0.708 - ETA: 1:04:58 - loss: 0.5406 - accuracy: 0.718 - ETA: 1:04:44 - loss: 0.5560 - accuracy: 0.713 - ETA: 1:04:27 - loss: 0.5645 - accuracy: 0.710 - ETA: 1:04:19 - loss: 0.5651 - accuracy: 0.711 - ETA: 1:04:07 - loss: 0.5586 - accuracy: 0.714 - ETA: 1:03:46 - loss: 0.5489 - accuracy: 0.722 - ETA: 1:03:30 - loss: 0.5460 - accuracy: 0.726 - ETA: 1:03:14 - loss: 0.5427 - accuracy: 0.729 - ETA: 1:02:58 - loss: 0.5602 - accuracy: 0.715 - ETA: 1:02:39 - loss: 0.5575 - accuracy: 0.720 - ETA: 1:02:25 - loss: 0.5584 - accuracy: 0

10000/10000 [==============================] - ETA: 28:38 - loss: 0.5553 - accuracy: 0.727 - ETA: 28:25 - loss: 0.5551 - accuracy: 0.726 - ETA: 28:13 - loss: 0.5552 - accuracy: 0.726 - ETA: 28:00 - loss: 0.5554 - accuracy: 0.726 - ETA: 27:48 - loss: 0.5550 - accuracy: 0.726 - ETA: 27:35 - loss: 0.5552 - accuracy: 0.726 - ETA: 27:22 - loss: 0.5560 - accuracy: 0.725 - ETA: 27:10 - loss: 0.5556 - accuracy: 0.725 - ETA: 26:57 - loss: 0.5558 - accuracy: 0.725 - ETA: 26:45 - loss: 0.5554 - accuracy: 0.726 - ETA: 26:32 - loss: 0.5557 - accuracy: 0.725 - ETA: 26:19 - loss: 0.5557 - accuracy: 0.725 - ETA: 26:07 - loss: 0.5566 - accuracy: 0.724 - ETA: 25:54 - loss: 0.5561 - accuracy: 0.725 - ETA: 25:42 - loss: 0.5567 - accuracy: 0.724 - ETA: 25:29 - loss: 0.5569 - accuracy: 0.724 - ETA: 25:16 - loss: 0.5569 - accuracy: 0.723 - ETA: 25:04 - loss: 0.5570 - accuracy: 0.723 - ETA: 24:51 - loss: 0.5575 - accuracy: 0.723 - ETA: 24:38 - loss: 0.5568 - accuracy: 0.723 - ETA: 24:25 - loss: 0.5562 - accur

 5632/10000 [===============>..............] - ETA: 1:05:02 - loss: 0.6044 - accuracy: 0.625 - ETA: 1:04:55 - loss: 0.4407 - accuracy: 0.781 - ETA: 1:04:37 - loss: 0.4628 - accuracy: 0.750 - ETA: 1:04:04 - loss: 0.5158 - accuracy: 0.726 - ETA: 1:04:02 - loss: 0.5373 - accuracy: 0.712 - ETA: 1:04:02 - loss: 0.5371 - accuracy: 0.713 - ETA: 1:04:11 - loss: 0.5492 - accuracy: 0.700 - ETA: 1:03:58 - loss: 0.5325 - accuracy: 0.707 - ETA: 1:03:47 - loss: 0.5358 - accuracy: 0.711 - ETA: 1:03:35 - loss: 0.5533 - accuracy: 0.706 - ETA: 1:03:26 - loss: 0.5534 - accuracy: 0.704 - ETA: 1:03:20 - loss: 0.5536 - accuracy: 0.705 - ETA: 1:03:07 - loss: 0.5413 - accuracy: 0.718 - ETA: 1:02:57 - loss: 0.5531 - accuracy: 0.709 - ETA: 1:02:46 - loss: 0.5586 - accuracy: 0.712 - ETA: 1:02:33 - loss: 0.5508 - accuracy: 0.718 - ETA: 1:02:19 - loss: 0.5587 - accuracy: 0.713 - ETA: 1:02:02 - loss: 0.5560 - accuracy: 0.717 - ETA: 1:01:48 - loss: 0.5565 - accuracy: 0.717 - ETA: 1:01:35 - loss: 0.5657 - accuracy: 0

10000/10000 [==============================] - ETA: 28:48 - loss: 0.5356 - accuracy: 0.731 - ETA: 28:35 - loss: 0.5355 - accuracy: 0.731 - ETA: 28:23 - loss: 0.5354 - accuracy: 0.731 - ETA: 28:10 - loss: 0.5356 - accuracy: 0.731 - ETA: 27:58 - loss: 0.5359 - accuracy: 0.730 - ETA: 27:45 - loss: 0.5356 - accuracy: 0.731 - ETA: 27:32 - loss: 0.5358 - accuracy: 0.731 - ETA: 27:20 - loss: 0.5357 - accuracy: 0.731 - ETA: 27:07 - loss: 0.5363 - accuracy: 0.730 - ETA: 26:55 - loss: 0.5357 - accuracy: 0.731 - ETA: 26:42 - loss: 0.5358 - accuracy: 0.731 - ETA: 26:29 - loss: 0.5352 - accuracy: 0.731 - ETA: 26:17 - loss: 0.5354 - accuracy: 0.731 - ETA: 26:04 - loss: 0.5351 - accuracy: 0.731 - ETA: 25:51 - loss: 0.5344 - accuracy: 0.732 - ETA: 25:39 - loss: 0.5345 - accuracy: 0.732 - ETA: 25:26 - loss: 0.5341 - accuracy: 0.732 - ETA: 25:14 - loss: 0.5346 - accuracy: 0.732 - ETA: 25:01 - loss: 0.5343 - accuracy: 0.731 - ETA: 24:49 - loss: 0.5340 - accuracy: 0.732 - ETA: 24:36 - loss: 0.5334 - accur

 5632/10000 [===============>..............] - ETA: 1:05:23 - loss: 0.4639 - accuracy: 0.781 - ETA: 1:05:30 - loss: 0.5566 - accuracy: 0.671 - ETA: 1:05:21 - loss: 0.5132 - accuracy: 0.718 - ETA: 1:04:49 - loss: 0.5144 - accuracy: 0.710 - ETA: 1:04:38 - loss: 0.5269 - accuracy: 0.725 - ETA: 1:04:24 - loss: 0.5055 - accuracy: 0.750 - ETA: 1:04:10 - loss: 0.5287 - accuracy: 0.741 - ETA: 1:04:05 - loss: 0.5459 - accuracy: 0.726 - ETA: 1:03:51 - loss: 0.5482 - accuracy: 0.732 - ETA: 1:03:45 - loss: 0.5344 - accuracy: 0.743 - ETA: 1:03:32 - loss: 0.5429 - accuracy: 0.747 - ETA: 1:03:21 - loss: 0.5337 - accuracy: 0.747 - ETA: 1:03:07 - loss: 0.5284 - accuracy: 0.747 - ETA: 1:02:50 - loss: 0.5207 - accuracy: 0.752 - ETA: 1:02:41 - loss: 0.5241 - accuracy: 0.752 - ETA: 1:02:34 - loss: 0.5270 - accuracy: 0.748 - ETA: 1:02:23 - loss: 0.5279 - accuracy: 0.744 - ETA: 1:02:10 - loss: 0.5278 - accuracy: 0.743 - ETA: 1:01:58 - loss: 0.5248 - accuracy: 0.743 - ETA: 1:01:46 - loss: 0.5212 - accuracy: 0

10000/10000 [==============================] - ETA: 28:41 - loss: 0.5288 - accuracy: 0.737 - ETA: 28:29 - loss: 0.5288 - accuracy: 0.737 - ETA: 28:16 - loss: 0.5282 - accuracy: 0.737 - ETA: 28:03 - loss: 0.5282 - accuracy: 0.737 - ETA: 27:51 - loss: 0.5286 - accuracy: 0.736 - ETA: 27:38 - loss: 0.5281 - accuracy: 0.737 - ETA: 27:26 - loss: 0.5284 - accuracy: 0.736 - ETA: 27:13 - loss: 0.5277 - accuracy: 0.736 - ETA: 27:00 - loss: 0.5274 - accuracy: 0.737 - ETA: 26:47 - loss: 0.5287 - accuracy: 0.736 - ETA: 26:35 - loss: 0.5289 - accuracy: 0.735 - ETA: 26:22 - loss: 0.5297 - accuracy: 0.735 - ETA: 26:09 - loss: 0.5291 - accuracy: 0.735 - ETA: 25:57 - loss: 0.5294 - accuracy: 0.735 - ETA: 25:44 - loss: 0.5301 - accuracy: 0.735 - ETA: 25:32 - loss: 0.5305 - accuracy: 0.734 - ETA: 25:19 - loss: 0.5304 - accuracy: 0.735 - ETA: 25:07 - loss: 0.5301 - accuracy: 0.735 - ETA: 24:54 - loss: 0.5302 - accuracy: 0.735 - ETA: 24:42 - loss: 0.5304 - accuracy: 0.734 - ETA: 24:29 - loss: 0.5300 - accur

 5632/10000 [===============>..............] - ETA: 1:04:38 - loss: 0.5276 - accuracy: 0.718 - ETA: 1:04:55 - loss: 0.5938 - accuracy: 0.671 - ETA: 1:04:54 - loss: 0.6065 - accuracy: 0.718 - ETA: 1:04:38 - loss: 0.5346 - accuracy: 0.765 - ETA: 1:04:31 - loss: 0.5376 - accuracy: 0.743 - ETA: 1:04:04 - loss: 0.5212 - accuracy: 0.750 - ETA: 1:03:57 - loss: 0.5099 - accuracy: 0.754 - ETA: 1:03:52 - loss: 0.5132 - accuracy: 0.742 - ETA: 1:03:39 - loss: 0.5142 - accuracy: 0.750 - ETA: 1:03:23 - loss: 0.5059 - accuracy: 0.756 - ETA: 1:03:06 - loss: 0.5212 - accuracy: 0.752 - ETA: 1:02:54 - loss: 0.5119 - accuracy: 0.755 - ETA: 1:02:47 - loss: 0.5068 - accuracy: 0.762 - ETA: 1:02:36 - loss: 0.5075 - accuracy: 0.758 - ETA: 1:02:39 - loss: 0.5012 - accuracy: 0.766 - ETA: 1:02:22 - loss: 0.5080 - accuracy: 0.765 - ETA: 1:02:10 - loss: 0.5067 - accuracy: 0.764 - ETA: 1:01:57 - loss: 0.5087 - accuracy: 0.763 - ETA: 1:01:45 - loss: 0.5160 - accuracy: 0.756 - ETA: 1:01:29 - loss: 0.5157 - accuracy: 0

10000/10000 [==============================] - ETA: 28:32 - loss: 0.5181 - accuracy: 0.748 - ETA: 28:20 - loss: 0.5182 - accuracy: 0.748 - ETA: 28:07 - loss: 0.5177 - accuracy: 0.748 - ETA: 27:55 - loss: 0.5178 - accuracy: 0.748 - ETA: 27:42 - loss: 0.5187 - accuracy: 0.748 - ETA: 27:30 - loss: 0.5185 - accuracy: 0.748 - ETA: 27:18 - loss: 0.5178 - accuracy: 0.749 - ETA: 27:05 - loss: 0.5176 - accuracy: 0.749 - ETA: 26:52 - loss: 0.5172 - accuracy: 0.749 - ETA: 26:40 - loss: 0.5172 - accuracy: 0.750 - ETA: 26:28 - loss: 0.5164 - accuracy: 0.750 - ETA: 26:15 - loss: 0.5154 - accuracy: 0.751 - ETA: 26:02 - loss: 0.5158 - accuracy: 0.751 - ETA: 25:50 - loss: 0.5159 - accuracy: 0.750 - ETA: 25:37 - loss: 0.5163 - accuracy: 0.750 - ETA: 25:25 - loss: 0.5161 - accuracy: 0.750 - ETA: 25:12 - loss: 0.5164 - accuracy: 0.750 - ETA: 25:00 - loss: 0.5169 - accuracy: 0.750 - ETA: 24:47 - loss: 0.5172 - accuracy: 0.750 - ETA: 24:35 - loss: 0.5174 - accuracy: 0.750 - ETA: 24:23 - loss: 0.5168 - accur

 5664/10000 [===============>..............] - ETA: 1:05:14 - loss: 0.5335 - accuracy: 0.718 - ETA: 1:04:56 - loss: 0.5290 - accuracy: 0.765 - ETA: 1:04:15 - loss: 0.5638 - accuracy: 0.729 - ETA: 1:04:22 - loss: 0.5366 - accuracy: 0.765 - ETA: 1:04:23 - loss: 0.5120 - accuracy: 0.775 - ETA: 1:04:06 - loss: 0.5242 - accuracy: 0.765 - ETA: 1:03:49 - loss: 0.5198 - accuracy: 0.767 - ETA: 1:03:24 - loss: 0.5136 - accuracy: 0.769 - ETA: 1:03:10 - loss: 0.5062 - accuracy: 0.763 - ETA: 1:02:56 - loss: 0.5003 - accuracy: 0.768 - ETA: 1:02:46 - loss: 0.5025 - accuracy: 0.772 - ETA: 1:02:31 - loss: 0.5011 - accuracy: 0.768 - ETA: 1:02:17 - loss: 0.4963 - accuracy: 0.769 - ETA: 1:02:14 - loss: 0.4978 - accuracy: 0.761 - ETA: 1:02:00 - loss: 0.5001 - accuracy: 0.756 - ETA: 1:01:47 - loss: 0.5067 - accuracy: 0.755 - ETA: 1:01:29 - loss: 0.4989 - accuracy: 0.762 - ETA: 1:01:16 - loss: 0.5002 - accuracy: 0.758 - ETA: 1:01:02 - loss: 0.4993 - accuracy: 0.758 - ETA: 1:00:49 - loss: 0.5015 - accuracy: 0

10000/10000 [==============================] - ETA: 28:16 - loss: 0.5139 - accuracy: 0.739 - ETA: 28:04 - loss: 0.5146 - accuracy: 0.739 - ETA: 27:52 - loss: 0.5144 - accuracy: 0.739 - ETA: 27:40 - loss: 0.5147 - accuracy: 0.738 - ETA: 27:27 - loss: 0.5137 - accuracy: 0.739 - ETA: 27:15 - loss: 0.5141 - accuracy: 0.738 - ETA: 27:02 - loss: 0.5144 - accuracy: 0.738 - ETA: 26:50 - loss: 0.5149 - accuracy: 0.738 - ETA: 26:37 - loss: 0.5157 - accuracy: 0.737 - ETA: 26:25 - loss: 0.5154 - accuracy: 0.738 - ETA: 26:12 - loss: 0.5161 - accuracy: 0.737 - ETA: 26:00 - loss: 0.5163 - accuracy: 0.737 - ETA: 25:47 - loss: 0.5184 - accuracy: 0.736 - ETA: 25:34 - loss: 0.5184 - accuracy: 0.736 - ETA: 25:22 - loss: 0.5185 - accuracy: 0.736 - ETA: 25:09 - loss: 0.5186 - accuracy: 0.736 - ETA: 24:57 - loss: 0.5183 - accuracy: 0.736 - ETA: 24:45 - loss: 0.5181 - accuracy: 0.736 - ETA: 24:32 - loss: 0.5182 - accuracy: 0.736 - ETA: 24:20 - loss: 0.5186 - accuracy: 0.736 - ETA: 24:07 - loss: 0.5190 - accur

 5632/10000 [===============>..............] - ETA: 1:06:18 - loss: 0.5704 - accuracy: 0.781 - ETA: 1:05:20 - loss: 0.5665 - accuracy: 0.765 - ETA: 1:04:31 - loss: 0.5721 - accuracy: 0.750 - ETA: 1:04:21 - loss: 0.5787 - accuracy: 0.742 - ETA: 1:04:23 - loss: 0.5664 - accuracy: 0.743 - ETA: 1:04:07 - loss: 0.5572 - accuracy: 0.744 - ETA: 1:03:46 - loss: 0.5488 - accuracy: 0.745 - ETA: 1:03:31 - loss: 0.5496 - accuracy: 0.753 - ETA: 1:03:18 - loss: 0.5454 - accuracy: 0.756 - ETA: 1:03:10 - loss: 0.5516 - accuracy: 0.746 - ETA: 1:03:05 - loss: 0.5497 - accuracy: 0.750 - ETA: 1:02:44 - loss: 0.5393 - accuracy: 0.747 - ETA: 1:02:39 - loss: 0.5345 - accuracy: 0.750 - ETA: 1:02:28 - loss: 0.5269 - accuracy: 0.750 - ETA: 1:02:13 - loss: 0.5290 - accuracy: 0.747 - ETA: 1:01:57 - loss: 0.5188 - accuracy: 0.752 - ETA: 1:01:41 - loss: 0.5234 - accuracy: 0.746 - ETA: 1:01:30 - loss: 0.5332 - accuracy: 0.743 - ETA: 1:01:30 - loss: 0.5341 - accuracy: 0.736 - ETA: 1:01:24 - loss: 0.5337 - accuracy: 0

10000/10000 [==============================] - ETA: 28:45 - loss: 0.5223 - accuracy: 0.739 - ETA: 28:32 - loss: 0.5222 - accuracy: 0.739 - ETA: 28:20 - loss: 0.5234 - accuracy: 0.739 - ETA: 28:07 - loss: 0.5236 - accuracy: 0.739 - ETA: 27:54 - loss: 0.5226 - accuracy: 0.740 - ETA: 27:42 - loss: 0.5229 - accuracy: 0.740 - ETA: 27:29 - loss: 0.5238 - accuracy: 0.739 - ETA: 27:16 - loss: 0.5239 - accuracy: 0.740 - ETA: 27:04 - loss: 0.5237 - accuracy: 0.740 - ETA: 26:51 - loss: 0.5226 - accuracy: 0.740 - ETA: 26:38 - loss: 0.5224 - accuracy: 0.740 - ETA: 26:26 - loss: 0.5218 - accuracy: 0.740 - ETA: 26:13 - loss: 0.5216 - accuracy: 0.741 - ETA: 26:00 - loss: 0.5219 - accuracy: 0.741 - ETA: 25:47 - loss: 0.5222 - accuracy: 0.740 - ETA: 25:35 - loss: 0.5228 - accuracy: 0.740 - ETA: 25:22 - loss: 0.5230 - accuracy: 0.740 - ETA: 25:09 - loss: 0.5227 - accuracy: 0.740 - ETA: 24:56 - loss: 0.5226 - accuracy: 0.740 - ETA: 24:44 - loss: 0.5232 - accuracy: 0.740 - ETA: 24:31 - loss: 0.5236 - accur

 5632/10000 [===============>..............] - ETA: 1:09:34 - loss: 0.5259 - accuracy: 0.781 - ETA: 1:06:46 - loss: 0.4930 - accuracy: 0.796 - ETA: 1:06:07 - loss: 0.4934 - accuracy: 0.781 - ETA: 1:05:51 - loss: 0.5014 - accuracy: 0.765 - ETA: 1:05:46 - loss: 0.5074 - accuracy: 0.756 - ETA: 1:05:25 - loss: 0.5267 - accuracy: 0.729 - ETA: 1:05:08 - loss: 0.5389 - accuracy: 0.723 - ETA: 1:04:48 - loss: 0.5314 - accuracy: 0.730 - ETA: 1:04:30 - loss: 0.5268 - accuracy: 0.736 - ETA: 1:04:21 - loss: 0.5502 - accuracy: 0.709 - ETA: 1:04:13 - loss: 0.5449 - accuracy: 0.707 - ETA: 1:03:52 - loss: 0.5417 - accuracy: 0.710 - ETA: 1:03:38 - loss: 0.5377 - accuracy: 0.713 - ETA: 1:03:26 - loss: 0.5496 - accuracy: 0.700 - ETA: 1:03:16 - loss: 0.5476 - accuracy: 0.704 - ETA: 1:02:58 - loss: 0.5401 - accuracy: 0.709 - ETA: 1:02:42 - loss: 0.5445 - accuracy: 0.709 - ETA: 1:02:29 - loss: 0.5361 - accuracy: 0.717 - ETA: 1:02:13 - loss: 0.5307 - accuracy: 0.720 - ETA: 1:01:58 - loss: 0.5321 - accuracy: 0

10000/10000 [==============================] - ETA: 29:10 - loss: 0.5212 - accuracy: 0.744 - ETA: 28:57 - loss: 0.5206 - accuracy: 0.744 - ETA: 28:44 - loss: 0.5198 - accuracy: 0.745 - ETA: 28:31 - loss: 0.5199 - accuracy: 0.745 - ETA: 28:19 - loss: 0.5192 - accuracy: 0.745 - ETA: 28:06 - loss: 0.5185 - accuracy: 0.746 - ETA: 27:53 - loss: 0.5191 - accuracy: 0.746 - ETA: 27:40 - loss: 0.5190 - accuracy: 0.746 - ETA: 27:27 - loss: 0.5188 - accuracy: 0.746 - ETA: 27:14 - loss: 0.5187 - accuracy: 0.746 - ETA: 27:02 - loss: 0.5179 - accuracy: 0.746 - ETA: 26:49 - loss: 0.5173 - accuracy: 0.746 - ETA: 26:36 - loss: 0.5170 - accuracy: 0.747 - ETA: 26:23 - loss: 0.5163 - accuracy: 0.747 - ETA: 26:11 - loss: 0.5160 - accuracy: 0.747 - ETA: 25:58 - loss: 0.5160 - accuracy: 0.747 - ETA: 25:45 - loss: 0.5160 - accuracy: 0.747 - ETA: 25:32 - loss: 0.5156 - accuracy: 0.747 - ETA: 25:20 - loss: 0.5153 - accuracy: 0.748 - ETA: 25:07 - loss: 0.5158 - accuracy: 0.747 - ETA: 24:54 - loss: 0.5154 - accur

 5632/10000 [===============>..............] - ETA: 1:05:01 - loss: 0.4646 - accuracy: 0.843 - ETA: 1:04:54 - loss: 0.5112 - accuracy: 0.750 - ETA: 1:04:17 - loss: 0.4764 - accuracy: 0.781 - ETA: 1:04:18 - loss: 0.5031 - accuracy: 0.757 - ETA: 1:04:10 - loss: 0.4801 - accuracy: 0.781 - ETA: 1:04:12 - loss: 0.4822 - accuracy: 0.781 - ETA: 1:04:11 - loss: 0.5066 - accuracy: 0.767 - ETA: 1:03:57 - loss: 0.5022 - accuracy: 0.773 - ETA: 1:03:44 - loss: 0.5007 - accuracy: 0.774 - ETA: 1:03:33 - loss: 0.5078 - accuracy: 0.765 - ETA: 1:03:19 - loss: 0.4978 - accuracy: 0.767 - ETA: 1:03:01 - loss: 0.5075 - accuracy: 0.752 - ETA: 1:02:50 - loss: 0.5102 - accuracy: 0.750 - ETA: 1:02:39 - loss: 0.5130 - accuracy: 0.747 - ETA: 1:02:38 - loss: 0.5158 - accuracy: 0.754 - ETA: 1:02:26 - loss: 0.5101 - accuracy: 0.753 - ETA: 1:02:06 - loss: 0.5074 - accuracy: 0.753 - ETA: 1:01:57 - loss: 0.5083 - accuracy: 0.751 - ETA: 1:01:46 - loss: 0.5153 - accuracy: 0.746 - ETA: 1:01:34 - loss: 0.5089 - accuracy: 0

10000/10000 [==============================] - ETA: 28:39 - loss: 0.5044 - accuracy: 0.750 - ETA: 28:27 - loss: 0.5045 - accuracy: 0.750 - ETA: 28:14 - loss: 0.5041 - accuracy: 0.750 - ETA: 28:02 - loss: 0.5034 - accuracy: 0.751 - ETA: 27:49 - loss: 0.5036 - accuracy: 0.751 - ETA: 27:37 - loss: 0.5033 - accuracy: 0.751 - ETA: 27:24 - loss: 0.5030 - accuracy: 0.751 - ETA: 27:12 - loss: 0.5035 - accuracy: 0.751 - ETA: 26:59 - loss: 0.5040 - accuracy: 0.751 - ETA: 26:46 - loss: 0.5046 - accuracy: 0.750 - ETA: 26:34 - loss: 0.5047 - accuracy: 0.751 - ETA: 26:21 - loss: 0.5043 - accuracy: 0.750 - ETA: 26:09 - loss: 0.5050 - accuracy: 0.750 - ETA: 25:57 - loss: 0.5045 - accuracy: 0.750 - ETA: 25:44 - loss: 0.5040 - accuracy: 0.751 - ETA: 25:32 - loss: 0.5043 - accuracy: 0.750 - ETA: 25:19 - loss: 0.5051 - accuracy: 0.749 - ETA: 25:06 - loss: 0.5046 - accuracy: 0.749 - ETA: 24:54 - loss: 0.5040 - accuracy: 0.749 - ETA: 24:41 - loss: 0.5036 - accuracy: 0.749 - ETA: 24:29 - loss: 0.5037 - accur

 5632/10000 [===============>..............] - ETA: 1:06:06 - loss: 0.5120 - accuracy: 0.687 - ETA: 1:06:00 - loss: 0.6010 - accuracy: 0.703 - ETA: 1:05:39 - loss: 0.6019 - accuracy: 0.687 - ETA: 1:05:32 - loss: 0.5452 - accuracy: 0.734 - ETA: 1:05:28 - loss: 0.5295 - accuracy: 0.731 - ETA: 1:05:10 - loss: 0.5352 - accuracy: 0.729 - ETA: 1:04:52 - loss: 0.5567 - accuracy: 0.709 - ETA: 1:04:28 - loss: 0.5567 - accuracy: 0.722 - ETA: 1:04:09 - loss: 0.5335 - accuracy: 0.739 - ETA: 1:03:57 - loss: 0.5232 - accuracy: 0.740 - ETA: 1:04:02 - loss: 0.5137 - accuracy: 0.744 - ETA: 1:03:51 - loss: 0.5148 - accuracy: 0.739 - ETA: 1:03:29 - loss: 0.5149 - accuracy: 0.740 - ETA: 1:03:15 - loss: 0.5182 - accuracy: 0.732 - ETA: 1:03:01 - loss: 0.5226 - accuracy: 0.731 - ETA: 1:02:45 - loss: 0.5263 - accuracy: 0.732 - ETA: 1:02:33 - loss: 0.5268 - accuracy: 0.733 - ETA: 1:02:14 - loss: 0.5273 - accuracy: 0.736 - ETA: 1:02:03 - loss: 0.5176 - accuracy: 0.743 - ETA: 1:01:52 - loss: 0.5132 - accuracy: 0

10000/10000 [==============================] - ETA: 28:54 - loss: 0.5040 - accuracy: 0.749 - ETA: 28:42 - loss: 0.5044 - accuracy: 0.749 - ETA: 28:29 - loss: 0.5040 - accuracy: 0.750 - ETA: 28:16 - loss: 0.5042 - accuracy: 0.749 - ETA: 28:03 - loss: 0.5040 - accuracy: 0.750 - ETA: 27:51 - loss: 0.5044 - accuracy: 0.750 - ETA: 27:39 - loss: 0.5053 - accuracy: 0.749 - ETA: 27:26 - loss: 0.5052 - accuracy: 0.749 - ETA: 27:13 - loss: 0.5064 - accuracy: 0.748 - ETA: 27:01 - loss: 0.5065 - accuracy: 0.748 - ETA: 26:48 - loss: 0.5064 - accuracy: 0.748 - ETA: 26:36 - loss: 0.5071 - accuracy: 0.748 - ETA: 26:23 - loss: 0.5066 - accuracy: 0.748 - ETA: 26:10 - loss: 0.5057 - accuracy: 0.748 - ETA: 25:57 - loss: 0.5062 - accuracy: 0.748 - ETA: 25:45 - loss: 0.5060 - accuracy: 0.748 - ETA: 25:32 - loss: 0.5059 - accuracy: 0.748 - ETA: 25:19 - loss: 0.5060 - accuracy: 0.748 - ETA: 25:07 - loss: 0.5056 - accuracy: 0.748 - ETA: 24:54 - loss: 0.5058 - accuracy: 0.747 - ETA: 24:41 - loss: 0.5060 - accur

 5632/10000 [===============>..............] - ETA: 1:04:52 - loss: 0.5712 - accuracy: 0.687 - ETA: 1:04:40 - loss: 0.5209 - accuracy: 0.781 - ETA: 1:03:56 - loss: 0.4826 - accuracy: 0.770 - ETA: 1:03:55 - loss: 0.4960 - accuracy: 0.765 - ETA: 1:03:44 - loss: 0.5143 - accuracy: 0.750 - ETA: 1:03:39 - loss: 0.5373 - accuracy: 0.734 - ETA: 1:03:18 - loss: 0.5148 - accuracy: 0.754 - ETA: 1:03:03 - loss: 0.5198 - accuracy: 0.742 - ETA: 1:02:54 - loss: 0.5150 - accuracy: 0.743 - ETA: 1:02:43 - loss: 0.5191 - accuracy: 0.740 - ETA: 1:02:32 - loss: 0.5217 - accuracy: 0.744 - ETA: 1:02:12 - loss: 0.5218 - accuracy: 0.744 - ETA: 1:02:06 - loss: 0.5121 - accuracy: 0.757 - ETA: 1:02:00 - loss: 0.5065 - accuracy: 0.761 - ETA: 1:01:50 - loss: 0.5091 - accuracy: 0.760 - ETA: 1:01:42 - loss: 0.5173 - accuracy: 0.753 - ETA: 1:01:34 - loss: 0.5203 - accuracy: 0.748 - ETA: 1:01:22 - loss: 0.5173 - accuracy: 0.748 - ETA: 1:01:23 - loss: 0.5142 - accuracy: 0.753 - ETA: 1:01:11 - loss: 0.5155 - accuracy: 0

10000/10000 [==============================] - ETA: 28:35 - loss: 0.4964 - accuracy: 0.757 - ETA: 28:22 - loss: 0.4956 - accuracy: 0.758 - ETA: 28:10 - loss: 0.4967 - accuracy: 0.757 - ETA: 27:57 - loss: 0.4959 - accuracy: 0.757 - ETA: 27:44 - loss: 0.4962 - accuracy: 0.757 - ETA: 27:32 - loss: 0.4960 - accuracy: 0.757 - ETA: 27:19 - loss: 0.4970 - accuracy: 0.757 - ETA: 27:06 - loss: 0.4971 - accuracy: 0.757 - ETA: 26:53 - loss: 0.4975 - accuracy: 0.756 - ETA: 26:41 - loss: 0.4970 - accuracy: 0.757 - ETA: 26:28 - loss: 0.4971 - accuracy: 0.757 - ETA: 26:15 - loss: 0.4977 - accuracy: 0.757 - ETA: 26:02 - loss: 0.4974 - accuracy: 0.757 - ETA: 25:50 - loss: 0.4979 - accuracy: 0.756 - ETA: 25:37 - loss: 0.4977 - accuracy: 0.756 - ETA: 25:24 - loss: 0.4971 - accuracy: 0.756 - ETA: 25:12 - loss: 0.4977 - accuracy: 0.755 - ETA: 24:59 - loss: 0.4973 - accuracy: 0.756 - ETA: 24:46 - loss: 0.4983 - accuracy: 0.755 - ETA: 24:34 - loss: 0.4988 - accuracy: 0.755 - ETA: 24:21 - loss: 0.4984 - accur

 5632/10000 [===============>..............] - ETA: 1:07:30 - loss: 0.7878 - accuracy: 0.593 - ETA: 1:06:38 - loss: 0.6096 - accuracy: 0.687 - ETA: 1:06:11 - loss: 0.5422 - accuracy: 0.729 - ETA: 1:06:08 - loss: 0.5318 - accuracy: 0.750 - ETA: 1:06:02 - loss: 0.5106 - accuracy: 0.775 - ETA: 1:06:57 - loss: 0.5031 - accuracy: 0.776 - ETA: 1:06:21 - loss: 0.5105 - accuracy: 0.763 - ETA: 1:06:01 - loss: 0.5076 - accuracy: 0.773 - ETA: 1:05:36 - loss: 0.5127 - accuracy: 0.767 - ETA: 1:05:04 - loss: 0.5047 - accuracy: 0.768 - ETA: 1:04:43 - loss: 0.4995 - accuracy: 0.767 - ETA: 1:04:29 - loss: 0.4932 - accuracy: 0.770 - ETA: 1:04:08 - loss: 0.4919 - accuracy: 0.762 - ETA: 1:03:50 - loss: 0.4839 - accuracy: 0.767 - ETA: 1:03:38 - loss: 0.4888 - accuracy: 0.762 - ETA: 1:03:32 - loss: 0.4892 - accuracy: 0.761 - ETA: 1:03:23 - loss: 0.4799 - accuracy: 0.772 - ETA: 1:03:13 - loss: 0.4889 - accuracy: 0.769 - ETA: 1:02:57 - loss: 0.4878 - accuracy: 0.764 - ETA: 1:02:39 - loss: 0.4859 - accuracy: 0

10000/10000 [==============================] - ETA: 30:24 - loss: 0.4958 - accuracy: 0.755 - ETA: 30:11 - loss: 0.4957 - accuracy: 0.755 - ETA: 29:57 - loss: 0.4950 - accuracy: 0.756 - ETA: 29:44 - loss: 0.4954 - accuracy: 0.755 - ETA: 29:31 - loss: 0.4948 - accuracy: 0.756 - ETA: 29:17 - loss: 0.4953 - accuracy: 0.756 - ETA: 29:03 - loss: 0.4950 - accuracy: 0.756 - ETA: 28:49 - loss: 0.4946 - accuracy: 0.757 - ETA: 28:37 - loss: 0.4945 - accuracy: 0.757 - ETA: 28:23 - loss: 0.4939 - accuracy: 0.757 - ETA: 28:11 - loss: 0.4936 - accuracy: 0.757 - ETA: 27:58 - loss: 0.4940 - accuracy: 0.757 - ETA: 27:45 - loss: 0.4935 - accuracy: 0.757 - ETA: 27:32 - loss: 0.4937 - accuracy: 0.757 - ETA: 27:18 - loss: 0.4935 - accuracy: 0.757 - ETA: 27:04 - loss: 0.4940 - accuracy: 0.757 - ETA: 26:51 - loss: 0.4935 - accuracy: 0.757 - ETA: 26:37 - loss: 0.4934 - accuracy: 0.757 - ETA: 26:23 - loss: 0.4933 - accuracy: 0.757 - ETA: 26:09 - loss: 0.4926 - accuracy: 0.757 - ETA: 25:55 - loss: 0.4929 - accur

 5632/10000 [===============>..............] - ETA: 1:08:07 - loss: 0.5380 - accuracy: 0.718 - ETA: 1:07:42 - loss: 0.5501 - accuracy: 0.781 - ETA: 1:06:56 - loss: 0.5460 - accuracy: 0.750 - ETA: 1:07:26 - loss: 0.5865 - accuracy: 0.718 - ETA: 1:07:33 - loss: 0.5889 - accuracy: 0.700 - ETA: 1:07:35 - loss: 0.5624 - accuracy: 0.724 - ETA: 1:07:31 - loss: 0.5692 - accuracy: 0.732 - ETA: 1:07:26 - loss: 0.5602 - accuracy: 0.750 - ETA: 1:07:58 - loss: 0.5422 - accuracy: 0.763 - ETA: 1:07:53 - loss: 0.5330 - accuracy: 0.768 - ETA: 1:07:51 - loss: 0.5425 - accuracy: 0.761 - ETA: 1:08:45 - loss: 0.5524 - accuracy: 0.752 - ETA: 1:09:37 - loss: 0.5489 - accuracy: 0.745 - ETA: 1:09:37 - loss: 0.5427 - accuracy: 0.752 - ETA: 1:09:17 - loss: 0.5295 - accuracy: 0.760 - ETA: 1:09:08 - loss: 0.5168 - accuracy: 0.767 - ETA: 1:09:26 - loss: 0.5195 - accuracy: 0.761 - ETA: 1:09:40 - loss: 0.5155 - accuracy: 0.760 - ETA: 1:09:14 - loss: 0.5110 - accuracy: 0.763 - ETA: 1:08:41 - loss: 0.5063 - accuracy: 0

10000/10000 [==============================] - ETA: 29:19 - loss: 0.4885 - accuracy: 0.764 - ETA: 29:06 - loss: 0.4888 - accuracy: 0.764 - ETA: 28:53 - loss: 0.4881 - accuracy: 0.765 - ETA: 28:39 - loss: 0.4878 - accuracy: 0.765 - ETA: 28:26 - loss: 0.4879 - accuracy: 0.764 - ETA: 28:13 - loss: 0.4875 - accuracy: 0.765 - ETA: 28:01 - loss: 0.4890 - accuracy: 0.765 - ETA: 27:48 - loss: 0.4897 - accuracy: 0.764 - ETA: 27:34 - loss: 0.4899 - accuracy: 0.763 - ETA: 27:21 - loss: 0.4911 - accuracy: 0.762 - ETA: 27:08 - loss: 0.4909 - accuracy: 0.763 - ETA: 26:55 - loss: 0.4904 - accuracy: 0.763 - ETA: 26:42 - loss: 0.4898 - accuracy: 0.763 - ETA: 26:29 - loss: 0.4896 - accuracy: 0.764 - ETA: 26:16 - loss: 0.4894 - accuracy: 0.764 - ETA: 26:03 - loss: 0.4904 - accuracy: 0.763 - ETA: 25:50 - loss: 0.4911 - accuracy: 0.762 - ETA: 25:37 - loss: 0.4912 - accuracy: 0.762 - ETA: 25:24 - loss: 0.4907 - accuracy: 0.763 - ETA: 25:10 - loss: 0.4904 - accuracy: 0.762 - ETA: 24:57 - loss: 0.4903 - accur

 5632/10000 [===============>..............] - ETA: 1:06:14 - loss: 0.4527 - accuracy: 0.781 - ETA: 1:06:11 - loss: 0.4670 - accuracy: 0.781 - ETA: 1:05:51 - loss: 0.5142 - accuracy: 0.781 - ETA: 1:05:19 - loss: 0.5016 - accuracy: 0.804 - ETA: 1:05:19 - loss: 0.5010 - accuracy: 0.787 - ETA: 1:05:22 - loss: 0.5194 - accuracy: 0.770 - ETA: 1:05:23 - loss: 0.5066 - accuracy: 0.772 - ETA: 1:05:17 - loss: 0.5157 - accuracy: 0.765 - ETA: 1:04:54 - loss: 0.4981 - accuracy: 0.774 - ETA: 1:04:37 - loss: 0.4999 - accuracy: 0.768 - ETA: 1:04:22 - loss: 0.5152 - accuracy: 0.750 - ETA: 1:04:10 - loss: 0.5263 - accuracy: 0.739 - ETA: 1:03:56 - loss: 0.5200 - accuracy: 0.742 - ETA: 1:03:42 - loss: 0.5111 - accuracy: 0.750 - ETA: 1:03:29 - loss: 0.5106 - accuracy: 0.747 - ETA: 1:03:17 - loss: 0.5119 - accuracy: 0.742 - ETA: 1:03:04 - loss: 0.5089 - accuracy: 0.744 - ETA: 1:02:53 - loss: 0.5023 - accuracy: 0.746 - ETA: 1:02:37 - loss: 0.5004 - accuracy: 0.751 - ETA: 1:02:23 - loss: 0.4961 - accuracy: 0

10000/10000 [==============================] - ETA: 29:12 - loss: 0.4916 - accuracy: 0.762 - ETA: 28:59 - loss: 0.4917 - accuracy: 0.762 - ETA: 28:47 - loss: 0.4923 - accuracy: 0.762 - ETA: 28:34 - loss: 0.4916 - accuracy: 0.763 - ETA: 28:21 - loss: 0.4909 - accuracy: 0.763 - ETA: 28:08 - loss: 0.4913 - accuracy: 0.763 - ETA: 27:56 - loss: 0.4907 - accuracy: 0.764 - ETA: 27:43 - loss: 0.4900 - accuracy: 0.764 - ETA: 27:31 - loss: 0.4904 - accuracy: 0.764 - ETA: 27:18 - loss: 0.4911 - accuracy: 0.763 - ETA: 27:06 - loss: 0.4906 - accuracy: 0.764 - ETA: 26:53 - loss: 0.4906 - accuracy: 0.764 - ETA: 26:40 - loss: 0.4909 - accuracy: 0.763 - ETA: 26:27 - loss: 0.4904 - accuracy: 0.764 - ETA: 26:14 - loss: 0.4898 - accuracy: 0.764 - ETA: 26:02 - loss: 0.4897 - accuracy: 0.764 - ETA: 25:49 - loss: 0.4892 - accuracy: 0.764 - ETA: 25:36 - loss: 0.4889 - accuracy: 0.765 - ETA: 25:23 - loss: 0.4891 - accuracy: 0.764 - ETA: 25:10 - loss: 0.4895 - accuracy: 0.764 - ETA: 24:58 - loss: 0.4892 - accur

In [12]:
# Load the model with the best validation accuracy
newModel.load_weights('input/saved_models/transfer_models_SGD.weights.best.hdf5')

In [13]:
loop=0

match=0
nonmatch=0

for test_data in test_tensors:
    pred = newModel.predict(np.expand_dims(test_data, axis=0))
    prediction = np.argmax(pred)
    if (np.argmax(test_targets[loop], axis=0) == prediction):
        match = match + 1
    else:
        nonmatch = nonmatch+1
    loop = loop + 1
    print ("I = {} M = {} D = {} ".format (loop, match, nonmatch))


I = 1 M = 1 D = 0 
I = 2 M = 1 D = 1 
I = 3 M = 2 D = 1 
I = 4 M = 3 D = 1 
I = 5 M = 4 D = 1 
I = 6 M = 5 D = 1 
I = 7 M = 5 D = 2 
I = 8 M = 6 D = 2 
I = 9 M = 7 D = 2 
I = 10 M = 8 D = 2 
I = 11 M = 9 D = 2 
I = 12 M = 10 D = 2 
I = 13 M = 11 D = 2 
I = 14 M = 11 D = 3 
I = 15 M = 12 D = 3 
I = 16 M = 13 D = 3 
I = 17 M = 13 D = 4 
I = 18 M = 13 D = 5 
I = 19 M = 14 D = 5 
I = 20 M = 14 D = 6 
I = 21 M = 15 D = 6 
I = 22 M = 16 D = 6 
I = 23 M = 17 D = 6 
I = 24 M = 17 D = 7 
I = 25 M = 17 D = 8 
I = 26 M = 18 D = 8 
I = 27 M = 19 D = 8 
I = 28 M = 20 D = 8 
I = 29 M = 21 D = 8 
I = 30 M = 22 D = 8 
I = 31 M = 22 D = 9 
I = 32 M = 23 D = 9 
I = 33 M = 23 D = 10 
I = 34 M = 24 D = 10 
I = 35 M = 24 D = 11 
I = 36 M = 24 D = 12 
I = 37 M = 25 D = 12 
I = 38 M = 26 D = 12 
I = 39 M = 27 D = 12 
I = 40 M = 28 D = 12 
I = 41 M = 29 D = 12 
I = 42 M = 29 D = 13 
I = 43 M = 29 D = 14 
I = 44 M = 29 D = 15 
I = 45 M = 29 D = 16 
I = 46 M = 30 D = 16 
I = 47 M = 31 D = 16 
I = 48 M = 32 D = 

I = 353 M = 240 D = 113 
I = 354 M = 241 D = 113 
I = 355 M = 242 D = 113 
I = 356 M = 243 D = 113 
I = 357 M = 244 D = 113 
I = 358 M = 245 D = 113 
I = 359 M = 246 D = 113 
I = 360 M = 247 D = 113 
I = 361 M = 247 D = 114 
I = 362 M = 248 D = 114 
I = 363 M = 248 D = 115 
I = 364 M = 249 D = 115 
I = 365 M = 249 D = 116 
I = 366 M = 250 D = 116 
I = 367 M = 251 D = 116 
I = 368 M = 251 D = 117 
I = 369 M = 252 D = 117 
I = 370 M = 253 D = 117 
I = 371 M = 253 D = 118 
I = 372 M = 254 D = 118 
I = 373 M = 255 D = 118 
I = 374 M = 256 D = 118 
I = 375 M = 257 D = 118 
I = 376 M = 258 D = 118 
I = 377 M = 258 D = 119 
I = 378 M = 259 D = 119 
I = 379 M = 260 D = 119 
I = 380 M = 261 D = 119 
I = 381 M = 262 D = 119 
I = 382 M = 262 D = 120 
I = 383 M = 262 D = 121 
I = 384 M = 263 D = 121 
I = 385 M = 263 D = 122 
I = 386 M = 264 D = 122 
I = 387 M = 265 D = 122 
I = 388 M = 266 D = 122 
I = 389 M = 267 D = 122 
I = 390 M = 268 D = 122 
I = 391 M = 269 D = 122 
I = 392 M = 270 D = 122 


I = 680 M = 478 D = 202 
I = 681 M = 478 D = 203 
I = 682 M = 479 D = 203 
I = 683 M = 480 D = 203 
I = 684 M = 481 D = 203 
I = 685 M = 482 D = 203 
I = 686 M = 483 D = 203 
I = 687 M = 484 D = 203 
I = 688 M = 485 D = 203 
I = 689 M = 486 D = 203 
I = 690 M = 487 D = 203 
I = 691 M = 488 D = 203 
I = 692 M = 489 D = 203 
I = 693 M = 489 D = 204 
I = 694 M = 490 D = 204 
I = 695 M = 491 D = 204 
I = 696 M = 492 D = 204 
I = 697 M = 492 D = 205 
I = 698 M = 493 D = 205 
I = 699 M = 494 D = 205 
I = 700 M = 495 D = 205 
I = 701 M = 496 D = 205 
I = 702 M = 497 D = 205 
I = 703 M = 498 D = 205 
I = 704 M = 499 D = 205 
I = 705 M = 500 D = 205 
I = 706 M = 500 D = 206 
I = 707 M = 501 D = 206 
I = 708 M = 502 D = 206 
I = 709 M = 503 D = 206 
I = 710 M = 504 D = 206 
I = 711 M = 505 D = 206 
I = 712 M = 506 D = 206 
I = 713 M = 506 D = 207 
I = 714 M = 507 D = 207 
I = 715 M = 508 D = 207 
I = 716 M = 508 D = 208 
I = 717 M = 509 D = 208 
I = 718 M = 510 D = 208 
I = 719 M = 511 D = 208 


I = 1007 M = 725 D = 282 
I = 1008 M = 726 D = 282 
I = 1009 M = 727 D = 282 
I = 1010 M = 728 D = 282 
I = 1011 M = 728 D = 283 
I = 1012 M = 729 D = 283 
I = 1013 M = 730 D = 283 
I = 1014 M = 731 D = 283 
I = 1015 M = 732 D = 283 
I = 1016 M = 733 D = 283 
I = 1017 M = 733 D = 284 
I = 1018 M = 734 D = 284 
I = 1019 M = 735 D = 284 
I = 1020 M = 735 D = 285 
I = 1021 M = 736 D = 285 
I = 1022 M = 737 D = 285 
I = 1023 M = 738 D = 285 
I = 1024 M = 739 D = 285 
I = 1025 M = 740 D = 285 
I = 1026 M = 741 D = 285 
I = 1027 M = 742 D = 285 
I = 1028 M = 743 D = 285 
I = 1029 M = 743 D = 286 
I = 1030 M = 744 D = 286 
I = 1031 M = 745 D = 286 
I = 1032 M = 746 D = 286 
I = 1033 M = 747 D = 286 
I = 1034 M = 748 D = 286 
I = 1035 M = 748 D = 287 
I = 1036 M = 748 D = 288 
I = 1037 M = 749 D = 288 
I = 1038 M = 750 D = 288 
I = 1039 M = 751 D = 288 
I = 1040 M = 751 D = 289 
I = 1041 M = 751 D = 290 
I = 1042 M = 751 D = 291 
I = 1043 M = 752 D = 291 
I = 1044 M = 753 D = 291 
I = 1045 M =

I = 1322 M = 966 D = 356 
I = 1323 M = 967 D = 356 
I = 1324 M = 968 D = 356 
I = 1325 M = 969 D = 356 
I = 1326 M = 970 D = 356 
I = 1327 M = 971 D = 356 
I = 1328 M = 971 D = 357 
I = 1329 M = 972 D = 357 
I = 1330 M = 973 D = 357 
I = 1331 M = 973 D = 358 
I = 1332 M = 974 D = 358 
I = 1333 M = 975 D = 358 
I = 1334 M = 976 D = 358 
I = 1335 M = 977 D = 358 
I = 1336 M = 978 D = 358 
I = 1337 M = 978 D = 359 
I = 1338 M = 979 D = 359 
I = 1339 M = 980 D = 359 
I = 1340 M = 981 D = 359 
I = 1341 M = 982 D = 359 
I = 1342 M = 983 D = 359 
I = 1343 M = 983 D = 360 
I = 1344 M = 983 D = 361 
I = 1345 M = 984 D = 361 
I = 1346 M = 985 D = 361 
I = 1347 M = 986 D = 361 
I = 1348 M = 986 D = 362 
I = 1349 M = 987 D = 362 
I = 1350 M = 987 D = 363 
I = 1351 M = 988 D = 363 
I = 1352 M = 989 D = 363 
I = 1353 M = 990 D = 363 
I = 1354 M = 991 D = 363 
I = 1355 M = 992 D = 363 
I = 1356 M = 993 D = 363 
I = 1357 M = 994 D = 363 
I = 1358 M = 994 D = 364 
I = 1359 M = 995 D = 364 
I = 1360 M =

I = 1627 M = 1198 D = 429 
I = 1628 M = 1199 D = 429 
I = 1629 M = 1200 D = 429 
I = 1630 M = 1201 D = 429 
I = 1631 M = 1202 D = 429 
I = 1632 M = 1203 D = 429 
I = 1633 M = 1203 D = 430 
I = 1634 M = 1204 D = 430 
I = 1635 M = 1205 D = 430 
I = 1636 M = 1206 D = 430 
I = 1637 M = 1207 D = 430 
I = 1638 M = 1207 D = 431 
I = 1639 M = 1208 D = 431 
I = 1640 M = 1209 D = 431 
I = 1641 M = 1210 D = 431 
I = 1642 M = 1211 D = 431 
I = 1643 M = 1212 D = 431 
I = 1644 M = 1213 D = 431 
I = 1645 M = 1214 D = 431 
I = 1646 M = 1215 D = 431 
I = 1647 M = 1216 D = 431 
I = 1648 M = 1217 D = 431 
I = 1649 M = 1218 D = 431 
I = 1650 M = 1219 D = 431 
I = 1651 M = 1220 D = 431 
I = 1652 M = 1220 D = 432 
I = 1653 M = 1221 D = 432 
I = 1654 M = 1221 D = 433 
I = 1655 M = 1222 D = 433 
I = 1656 M = 1223 D = 433 
I = 1657 M = 1224 D = 433 
I = 1658 M = 1225 D = 433 
I = 1659 M = 1226 D = 433 
I = 1660 M = 1226 D = 434 
I = 1661 M = 1227 D = 434 
I = 1662 M = 1228 D = 434 
I = 1663 M = 1229 D = 434 
I

I = 1930 M = 1427 D = 503 
I = 1931 M = 1428 D = 503 
I = 1932 M = 1429 D = 503 
I = 1933 M = 1430 D = 503 
I = 1934 M = 1430 D = 504 
I = 1935 M = 1431 D = 504 
I = 1936 M = 1432 D = 504 
I = 1937 M = 1432 D = 505 
I = 1938 M = 1433 D = 505 
I = 1939 M = 1434 D = 505 
I = 1940 M = 1435 D = 505 
I = 1941 M = 1436 D = 505 
I = 1942 M = 1436 D = 506 
I = 1943 M = 1437 D = 506 
I = 1944 M = 1438 D = 506 
I = 1945 M = 1439 D = 506 
I = 1946 M = 1439 D = 507 
I = 1947 M = 1440 D = 507 
I = 1948 M = 1441 D = 507 
I = 1949 M = 1442 D = 507 
I = 1950 M = 1443 D = 507 
I = 1951 M = 1444 D = 507 
I = 1952 M = 1445 D = 507 
I = 1953 M = 1446 D = 507 
I = 1954 M = 1446 D = 508 
I = 1955 M = 1446 D = 509 
I = 1956 M = 1446 D = 510 
I = 1957 M = 1447 D = 510 
I = 1958 M = 1447 D = 511 
I = 1959 M = 1447 D = 512 
I = 1960 M = 1448 D = 512 
I = 1961 M = 1448 D = 513 
I = 1962 M = 1449 D = 513 
I = 1963 M = 1450 D = 513 
I = 1964 M = 1451 D = 513 
I = 1965 M = 1452 D = 513 
I = 1966 M = 1452 D = 514 
I

I = 2233 M = 1651 D = 582 
I = 2234 M = 1652 D = 582 
I = 2235 M = 1652 D = 583 
I = 2236 M = 1652 D = 584 
I = 2237 M = 1653 D = 584 
I = 2238 M = 1654 D = 584 
I = 2239 M = 1654 D = 585 
I = 2240 M = 1655 D = 585 
I = 2241 M = 1656 D = 585 
I = 2242 M = 1657 D = 585 
I = 2243 M = 1657 D = 586 
I = 2244 M = 1657 D = 587 
I = 2245 M = 1657 D = 588 
I = 2246 M = 1657 D = 589 
I = 2247 M = 1658 D = 589 
I = 2248 M = 1659 D = 589 
I = 2249 M = 1660 D = 589 
I = 2250 M = 1661 D = 589 
I = 2251 M = 1661 D = 590 
I = 2252 M = 1662 D = 590 
I = 2253 M = 1663 D = 590 
I = 2254 M = 1664 D = 590 
I = 2255 M = 1664 D = 591 
I = 2256 M = 1664 D = 592 
I = 2257 M = 1665 D = 592 
I = 2258 M = 1666 D = 592 
I = 2259 M = 1667 D = 592 
I = 2260 M = 1667 D = 593 
I = 2261 M = 1668 D = 593 
I = 2262 M = 1668 D = 594 
I = 2263 M = 1669 D = 594 
I = 2264 M = 1669 D = 595 
I = 2265 M = 1670 D = 595 
I = 2266 M = 1671 D = 595 
I = 2267 M = 1672 D = 595 
I = 2268 M = 1673 D = 595 
I = 2269 M = 1673 D = 596 
I

In [14]:
# evaluate and print the test accuracy
# get index of predicted smile detection for each image in test set
smile_prediction = [np.argmax(newModel.predict(np.expand_dims(test_data, axis=0))) for test_data in test_tensors]

# report test accuracy
test_accuracy = 100*np.sum(np.array(smile_prediction)==np.argmax(test_targets, axis=1))/len(smile_prediction)
print('Test accuracy: %.4f%%' % test_accuracy)

Test accuracy: 74.2800%



Here, we see that an accuracy of 74.28% is achieved with this model.